plone/subrequest/__init__.py

```diff
@@ -78,8 +72,8 @@ class IDisableCSRFProtection(Interface):
 
 def subrequest(url, root=None, stdout=None, exception_handler=None):
     assert url is not None, 'You must pass a url'
-    if isinstance(url, six.binary_type):
-        url = url.decode('utf-8')
```

Comment:
Just trying to guess... 
Isn't this the line that caused the bug? I did not have time to check this out a lot but I believe that this should be a `str` in Python2. Note that it is later encoded again (line 107) and the old code was doing:
```
            if isinstance(parent_url, six.binary_type):
￼                parent_url = parent_url.encode('utf-8')
```

Reply:
That is probably the bad line yes. But maybe the difference is not on Python 2 versus 3, but in Zope 2 versus 4. Hard to say.
But the point of this PR is that none of the changes of 1.8.5 and 1.8.6 are needed or useful on the Plone versions that use it.